numpy/lib/tests/test_function_base.py

```diff
@@ -1499,6 +1500,60 @@ def test_size_zero_output(self):
         with assert_raises_regex(ValueError, 'new output dimensions'):
             f(x)
 
+    @pytest.mark.skipif(not HAS_REFCOUNT, reason="Python lacks refcounts")
+    def test_frompyfunc_leaks(self):
+        # exposed in gh-11867 as np.vectorized, but the problem stems from
+        # frompyfunc.
+        # class.attribute = np.frompyfunc(<method>) creates a
+        # reference cycle that requires a gc collection cycle to break
+        # (on CPython 3)
+        import gc
+        import types
+
+        gc.disable()
+        iters = 20
+        try:
+            class A(object):
+                def method(self, *args):
+                    return np.int(0)
+
+                @staticmethod
+                def method1(*args):
+                    return np.int(0)
+
+
+            refcount = sys.getrefcount(A.method)
+            # Bound method
+            for i in range(iters):
+                a = A()
+                a.f_bound = types.MethodType(np.frompyfunc(A.method, 1, 1), a)
+                out = a.f_bound(np.arange(10))
+            a = None
+            if PY2:
+                assert_equal(sys.getrefcount(A.method), refcount)
+            else:
+                # a.method refcount is part of a reference cycle
+                assert_equal(sys.getrefcount(A.method), refcount + iters)
+            for i in range(5):
+                gc.collect()
+            assert_equal(sys.getrefcount(A.method), refcount)
+
+            #Unbound method
+            for i in range(iters):
+                a = A()
+                a.f_unbound = np.frompyfunc(A.method1, 1, 1)
+                out = a.f_unbound(np.arange(10))
+            a = None
+            if PY2:
+                assert_equal(sys.getrefcount(A.method), refcount)
+            else:
+                # a.method refcount is part of a reference cycle
+                assert_equal(sys.getrefcount(A.method), refcount + iters)
```

Comment:
Shouldn't you be checking the refcount of `method1` here? You never even touch `method` in this half of the test.

Reply:
I will rewrite the tests to make sure they fail before/pass after this change